llama.cpp

```diff
@@ -5986,11 +6003,10 @@ static uint8_t llama_token_to_byte(const llama_vocab& vocab, llama_token id) {
 }
 
 static llama_token llama_byte_to_token(const llama_vocab & vocab, uint8_t ch) {
+    const char * hex = "0123456789ABCDEF";
     switch (llama_vocab_get_type(vocab)) {
     case LLAMA_VOCAB_TYPE_SPM: {
-        char buf[7];
-        int result = snprintf(buf, sizeof(buf), "<0x%02X>", ch);
-        GGML_ASSERT(0 <= result && result < 7);
+        const char buf[7] = { '<', '0', 'x', hex[ch >> 4], hex[ch & 15], '>', 0 };
```

Comment:
I believe you can just write `buf[]` here since the size is implied by the initializer.

Reply:
I thought there were some compilers that couldn't handle it, but I might be mistaken. I think leaving the single character here is worth it though just to be more explicit. I.E. if someone messes with that code in the future and accidentally removes the `0`, it should produce a compiler error. Not a likely scenario, but still...

Reply:
It won't produce a compiler error, it will just silently add zero bytes to the end of the array. Array and struct initializers are not required to initialize all elements.